packages/@tailwindcss-upgrade/src/template/candidates.ts

```diff
@@ -133,8 +134,75 @@ function printVariant(variant: Variant) {
   return base
 }
 
-function escapeArbitrary(input: string) {
-  return input
+function printArbitraryValue(input: string) {
+  let ast = ValueParser.parse(input)
+
+  let drop = new Set<ValueParser.ValueAstNode>()
+
+  ValueParser.walk(ast, (node, { parent }) => {
+    let parentArray = parent === null ? ast : (parent.nodes ?? [])
+
+    // Handle operators (e.g.: inside of `calc(…)`)
+    if (
+      node.kind === 'word' &&
+      // Operators
+      (node.value === '+' || node.value === '-' || node.value === '*' || node.value === '/')
+    ) {
+      let idx = parentArray.indexOf(node) ?? -1
+
+      // This should not be possible
+      if (idx === -1) return
+
+      let previous = parentArray[idx - 1]
+      if (previous?.kind !== 'separator' || previous.value !== ' ') return
```

Comment:
What actually happens when we return `undefined` here? We're still interpolating the return value of this function into the class name at the call site.

Reply:
The surrounding `walk()` will keep walking (equivalent to `WalkAction.Continue`) because it could be that you have multiple `calc(…)` calls for example and if we can't upgrade the first one, we might be able to upgrade the second one.